### ISSUE TITLE
[Enhancement]: tagging inline security group rules in existing resource or a new aws_security_group_v2

### ISSUE THREAD

**Issue by nitrocode** [hashicorp/terraform-provider-aws#37914]
### Description

Id like exclusive management of my security group rules to be in terraform and i want tagging.

There isnt an easy way to do this unless i want to forego exclusive management (vulnerable to clickops) or forego tagging.

Perhaps the original resource can be updated to allow tagging of inline rules or a new v2 resource can be created?

### Affected Resource(s) and/or Data Source(s)

* aws_security_group

### Potential Terraform Configuration

```terraform
Tbd
```

### Would you like to implement a fix?

No

**Comment by ewbankkit** [comment]
@nitrocode The [`aws_vpc_security_group_ingress_rule`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc_security_group_ingress_rule) and [`aws_vpc_security_group_egress_rule`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc_security_group_egress_rule) resources are now the preferred way to managed security group rules and they both support rule tagging.

**Comment by nitrocode** [comment]
Yes but they don't allow me to enforce the entire security group in code like the inline rules do. This is why I'm suggesting either a separate sg resource or a modification of the existing sg resource
